src/entities/Updates/routes.ts

```diff
@@ -0,0 +1,119 @@
+import RequestError from 'decentraland-gatsby/dist/entities/Route/error'
+import Time from 'decentraland-gatsby/dist/utils/date/Time'
+import { getNextUpdate, getPublicUpdates, getPendingUpdates, getCurrentUpdate, isBetweenLateThresholdDate } from './utils'
+import UpdateModel from './model'
+import { UpdateAttributes, UpdateStatus } from './types'
+import ProposalModel from '../Proposal/model'
+import { ProposalAttributes } from '../Proposal/types'
+
+export default routes((route) => {
+  const withAuth = auth()
+  route.get('/proposals/:proposal/updates', handleAPI(getProposalUpdates))
+  route.get('/proposals/:update/update', handleAPI(getProposalUpdate))
+  route.post('/proposals/:proposal/update', withAuth, handleAPI(createProposalUpdate))
+  route.patch('/proposals/:proposal/update', withAuth, handleAPI(updateProposalUpdate))
+})
+
+async function getProposalUpdate(req: Request<{ update: string }>) {
+  const id = req.params.update
+
+  if (!id) {
+    throw new RequestError(`Missing id`, RequestError.NotFound)
+  }
+
+  const update = await UpdateModel.findOne<UpdateAttributes>({ id })
+
+  if (!update) {
+    throw new RequestError(`Update not found: "${id}"`, RequestError.NotFound)
+  }
+
+  return update
+}
+
+async function getProposalUpdates(req: Request<{ proposal: string }>) {
+  const proposal_id = req.params.proposal
+
+  if (!proposal_id) {
+    throw new RequestError(`Proposal not found: "${proposal_id}"`, RequestError.NotFound)
+  }
+
+  const updates = await UpdateModel.find<UpdateAttributes>({ proposal_id })
+  const publicUpdates = getPublicUpdates(updates)
+  const nextUpdate = getNextUpdate(updates)
+  const currentUpdate = getCurrentUpdate(updates)
+  const pendingUpdates = getPendingUpdates(updates)
+
+  return {
+    publicUpdates,
+    pendingUpdates,
+    nextUpdate,
+    currentUpdate,
+  }
+}
+
+async function createProposalUpdate(req: WithAuth<Request<{ proposal: string }>>) {
+  const { health, introduction, highlights, blockers, next_steps, additional_notes } = req.body
+
+  const user = req.auth!
+  const proposalId = req.params.proposal
+  const proposal = await ProposalModel.findOne<ProposalAttributes>({ id: proposalId })
+
+  if (proposal?.user !== user) {
+    throw new RequestError(`Unauthorized`, RequestError.Forbidden)
+  }
+
+  const updates = await UpdateModel.find({
+    proposal_id: proposalId,
+    status: UpdateStatus.Pending
+  })
+
+  if (updates.length > 0) {
+    throw new RequestError(`Updates pending for this proposal`, RequestError.BadRequest)
+  }
+
+  return await UpdateModel.createUpdate({
+    proposal_id: proposal?.id!,
+    health,
+    introduction,
+    highlights,
+    blockers,
+    next_steps,
+    additional_notes,
+  })
+}
+
+async function updateProposalUpdate(req: WithAuth<Request<{ proposal: string }>>) {
+  const { id, health, introduction, highlights, blockers, next_steps, additional_notes } = req.body
+  const update = await UpdateModel.findOne(id)
```

Comment:
Here we are not checking if the result was successful. 

I'm thinking we often have this problem, and sometimes we check, sometimes we don't.

Maybe we could think of some kind of behavior we could add to the model classes, that could receive a callback for when the search is not successful. I'm thinking of an interface or a mixin, but there could be simpler ways.

```typescript
function findOneOrCall(id, callback){
         const value = this.findOne(id)
         if(!value) callback(id);
         return value
}
```

and have a frontend common callback for when something is not found, to return a `BadRequest` and the searched element/id.

It's just an idea. No need to implement anything right now.

Reply:
Added the check in this case, let's think how we can handle this later!